src/workspaces/SettingsModal/SettingsModal.tsx

```diff
@@ -0,0 +1,176 @@
+        .getSettings()) satisfies WorkspaceSetting[];
+      setWorkspaceSettings(settings);
+      const bucketLifecycleSetting = getFirstBucketLifecycleSetting(settings, true);
+      if (bucketLifecycleSetting !== undefined) {
+        const rule = getDeleteLifecycleRule(bucketLifecycleSetting, true);
+        if (rule) {
+          setLifecycleRulesEnabled(true);
+          const prefixes = rule.conditions.matchesPrefix;
+          if (prefixes.length === 0) {
+            setPrefixes([suggestedPrefixes.allObjects]);
+          } else {
+            setPrefixes(prefixes);
+          }
+          setLifecycleAge(rule.conditions.age);
+        }
+      }
+    });
+
+    loadSettings();
+  }, [namespace, name, signal]);
+
+  const persistSettings = withErrorReporting('Error saving workspace settings')(async () => {
+    let newSettings: WorkspaceSetting[];
+    if (lifecycleRulesEnabled) {
+      const prefixesOrNone = _.without([suggestedPrefixes.allObjects], prefixes);
+      newSettings = modifyFirstBucketDeletionRule(workspaceSettings || [], lifecycleAge!, prefixesOrNone);
+    } else {
+      newSettings = removeFirstBucketDeletionRule(workspaceSettings || []);
+    }
+    await Ajax().Workspaces.workspaceV2(namespace, name).updateSettings(newSettings);
+    props.onDismiss();
+    const originalLifecycleSetting = getFirstBucketLifecycleSetting(workspaceSettings || []);
+    const newLifecycleSetting = getFirstBucketLifecycleSetting(newSettings);
+    if (!_.isEqual(originalLifecycleSetting, newLifecycleSetting)) {
+      let prefixesChoice: string | null = null;
+      if (lifecycleRulesEnabled) {
+        if (_.without(_.values(suggestedPrefixes), prefixes).length > 0) {
+          prefixesChoice = 'Custom';
+        } else if (_.contains(suggestedPrefixes.allObjects, prefixes)) {
+          prefixesChoice = 'AllObjects';
+        } else {
+          const submissions = _.contains(suggestedPrefixes.submissions, prefixes);
+          const intermediaries = _.contains(suggestedPrefixes.submissionIntermediaries, prefixes);
+          if (submissions && intermediaries) {
+            prefixesChoice = 'AllSubmissionsAndSubmissionsIntermediaries';
+          } else if (submissions) {
+            prefixesChoice = 'AllSubmissions';
+          } else if (intermediaries) {
+            prefixesChoice = 'SubmissionsIntermediaries';
+          }
+        }
+      }
+      Ajax().Metrics.captureEvent(Events.workspaceSettingsBucketLifecycle, {
+        enabled: lifecycleRulesEnabled,
+        prefix: prefixesChoice,
+        age: lifecycleAge, // will be null if lifecycleRulesEnabled is false
+        ...extractWorkspaceDetails(props.workspace),
+      });
+    }
+  });
+
+  const getSaveTooltip = () => {
+    if (!isOwner) {
+      return 'You do not have permissions to modify settings';
+    }
+    if (lifecycleRulesEnabled && (prefixes.length === 0 || lifecycleAge === null)) {
+      return 'Please specify all lifecycle rule options';
```

Comment:
We will have to generalize further or add more conditions when this dialog also has soft delete.